homeassistant/components/sensor/flunearyou.py

```diff
@@ -0,0 +1,215 @@
+        (TYPE_USER_TOTAL, 'Total Symptoms', 'mdi:alert', 'reports'),
+    ]
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_LATITUDE):
+        cv.latitude,
+    vol.Optional(CONF_LONGITUDE):
+        cv.longitude,
+    vol.Required(CONF_MONITORED_CONDITIONS, default=list(SENSORS)):
+        vol.All(cv.ensure_list, [vol.In(SENSORS)])
+})
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Configure the platform and add the sensors."""
+    from pyflunearyou import create_client
+    from pyflunearyou.errors import FluNearYouError
+
+    websession = aiohttp_client.async_get_clientsession(hass)
+
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+    identifier = '{0},{1}'.format(latitude, longitude)
+
+    try:
+        client = await create_client(latitude, longitude, websession)
+    except FluNearYouError as err:
+        _LOGGER.error('There was an error while setting up: %s', err)
+        return
+
+    fny = FluNearYouData(client, config[CONF_MONITORED_CONDITIONS])
+    await fny.async_update()
+
+    sensors = [
+        FluNearYouSensor(fny, kind, name, identifier, category, icon, unit)
+        for category in config[CONF_MONITORED_CONDITIONS]
+        for kind, name, icon, unit in SENSORS[category]
+    ]
+
+    async_add_entities(sensors, True)
+
+
+class FluNearYouSensor(Entity):
+    """Define a base Flu Near You sensor."""
+
+    def __init__(self, fny, kind, name, identifier, category, icon, unit):
+        """Initialize the sensor."""
+        self._attrs = {ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION}
+        self._category = category
+        self._icon = icon
+        self._identifier = identifier
+        self._kind = kind
+        self._name = name
+        self._state = None
+        self._unit = unit
+        self.fny = fny
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return bool(self.fny.data[self._category])
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        return self._attrs
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return self._icon
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self._state
+
+    @property
+    def unique_id(self):
+        """Return a unique, HASS-friendly identifier for this entity."""
+        return '{0}_{1}'.format(self._identifier, self._kind)
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._unit
+
+    async def async_update(self):
+        """Update the sensor."""
+        await self.fny.async_update()
+
+        cdc_data = self.fny.data[CATEGORY_CDC_REPORT]
```

Comment:
Use `dict.get` here, otherwise there could be a `KeyError`.

Reply:
The `_get_data()` method sets the category to `{}` so I think this lookup is okay.

Reply:
Not for all monitored conditions, always.

Reply:
You are right – if it is not monitored, it is not updated.